velox/functions/prestosql/tests/ArrayDuplicatesTest.cpp

```diff
@@ -0,0 +1,241 @@
+    assertEqualVectors(expected, result);
+  }
+
+  // Execute test for integer types.
+  template <typename T>
+  void testInt() {
+    auto array = makeNullableArrayVector<T>({
+        {},
+        {0},
+        {1},
+        {std::numeric_limits<T>::min()},
+        {std::numeric_limits<T>::max()},
+        {std::nullopt},
+        {-1},
+        {1, 2, 3},
+        {1, 2, 1},
+        {1, 1, 1},
+        {-1, -2, -3},
+        {-1, -2, -1},
+        {-1, -1, -1},
+        {std::nullopt, std::nullopt, std::nullopt},
+        {1, 2, -2, 1},
+        {1, 1, -2, -2, -2, 4, 8},
+        {3, 8, std::nullopt},
+        {1, 2, 3, std::nullopt, 4, 1, 2, std::nullopt},
+    });
+
+    auto expected = makeNullableArrayVector<T>({
+        {},
+        {},
+        {},
+        {},
+        {},
+        {},
+        {},
+        {},
+        {1},
+        {1},
+        {},
+        {-1},
+        {-1},
+        {std::nullopt},
+        {1},
+        {1, -2},
+        {},
+        {1, 2, std::nullopt},
+    });
+
+    testExpr(expected, "array_duplicates(C0)", {array});
+  }
+};
+
+} // namespace
+
+// Test integer arrays.
+TEST_F(ArrayDuplicatesTest, integerArrays) {
+  testInt<int8_t>();
+  testInt<int16_t>();
+  testInt<int32_t>();
+  testInt<int64_t>();
+}
+
+// Test inline (short) strings.
+TEST_F(ArrayDuplicatesTest, inlineStringArrays) {
+  using S = StringView;
+
+  auto array = makeNullableArrayVector<StringView>({
+      {},
+      {S("")},
+      {S(" ")},
+      {S("a")},
+      {std::nullopt},
+      {S("a"), S("b")},
+      {S("a"), S("A")},
+      {S("a"), S("a")},
+      {std::nullopt, std::nullopt},
+      {S("a"), std::nullopt, S("b")},
+      {S("a"), S("b"), S("a"), S("a")},
+      {std::nullopt, S("b"), std::nullopt},
+      {S("abc")},
+  });
+
+  auto expected = makeNullableArrayVector<StringView>({
+      {},
+      {},
+      {},
+      {},
+      {},
+      {},
+      {},
+      {S("a")},
+      {std::nullopt},
+      {},
+      {S("a")},
+      {std::nullopt},
+      {},
+  });
+
+  testExpr(expected, "array_duplicates(C0)", {array});
+}
+
+// Test non-inline (> 12 character length) strings.
+TEST_F(ArrayDuplicatesTest, stringArrays) {
+  using S = StringView;
+
+  auto array = makeNullableArrayVector<StringView>({
+      {S("red shiny car ahead"), S("blue clear sky above")},
+      {std::nullopt,
+       S("blue clear sky above"),
+       S("yellow rose flowers"),
+       S("blue clear sky above"),
+       S("orange beautiful sunset")},
+      {std::nullopt, std::nullopt},
+      {},
+      {S("red shiny car ahead"),
+       S("purple is an elegant color"),
+       S("green plants make us happy")},
+  });
+
+  auto expected = makeNullableArrayVector<StringView>({
+      {},
+      {S("blue clear sky above")},
+      {std::nullopt},
+      {},
+      {},
+  });
+
+  testExpr(expected, "array_duplicates(C0)", {array});
+}
+
+TEST_F(ArrayDuplicatesTest, testNonContiguousRows) {
```

Comment:
Thanks for @mbasmanova's new test case to verify the fix on `rawSizes` and `rawOffsets` pointers above.

Reply:
naming: testNonContiguousRows -> nonContiguousRows